reader/views.py

```diff
@@ -2199,13 +2200,23 @@ def calendars_api(request):
         import datetime
         diaspora = request.GET.get("diaspora", "1")
         custom = request.GET.get("custom", None)
+
+        zone = request.GET.get("timezone", None)
+        if zone:
+            try:
+                zone = pytz.timezone(zone)
+            except pytz.exceptions.UnknownTimeZoneError as e:
+                return jsonResponse({"error": "Unknown 'timezone' value: '%s'." % zone})\
+        else:
+            zone = timezone.now()
```

Comment:
It seems like `zone` should be of type which is subclass of `tzinfo`. Here you're setting it to `datetime.datetime`. Pretty sure this is an error.

Reply:
Hrm, yeah, you're right. I'd like to have the exception case that uses the localtime to also consider the timezone. I think that reverting that line, and then taking your suggestion above should do the trick